flink-contrib/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBKeyedStateBackend.java

```diff
@@ -761,6 +769,13 @@ private void restoreKVStateMetaData() throws IOException, ClassNotFoundException
 		private void restoreKVStateData() throws IOException, RocksDBException {
 			//for all key-groups in the current state handle...
 			for (Tuple2<Integer, Long> keyGroupOffset : currentKeyGroupsStateHandle.getGroupRangeOffsets()) {
+				int keyGroup = keyGroupOffset.f0;
+
+				// Skip those key groups that do not belong to the backend
+				if (!rocksDBKeyedStateBackend.getKeyGroupRange().contains(keyGroup)) {
```

Comment:
The same comment I had on the `HeapKeyedStateBackend` also applies here: I think the post-filter is no longer required after the change i suggested for `StateAssignmentOperation`.